service/proxy/router/server.go

```diff
@@ -46,11 +48,11 @@ func Serve(router Router, replSvc replication.Service) *http.ServeMux {
 			}
 		}()
 		ctx = log.WithStorage(ctx, storage)
+		// TODO: is it reachable? This branch is active only if err == nil
```

Comment:
yes it is reachable

Reply:
I've added this comment because it looks suspicious.

`err != nil` check is performed on L39. If it is successful, then function exits.

However, on L52, we are checking error type and logging error itself.

If it works in some other way, it is worth to explain it.

Reply:
i think you are right. if i remember it correctly, if err!=nil, we can additionnaly check if it was s3 api error or some other error like network problem.
See: https://github.com/clyso/chorus/blob/main/pkg/s3client/http_client.go#L243-L263

Reply:
lets do later

Reply:
Made #67 to cover it.